client/state/add-reducer.ts

```diff
@@ -26,9 +26,9 @@ async function initializeState(
 // For a given store, creates a function that adds a new reducer to the store,
 // and loads (asynchronously) and applies the persisted state for it.
 export const addReducerToStore = ( store: object ) => (
-	key: string,
+	key: string[],
 	reducer: ( state: object, action: object ) => object
```

Comment:
We might be able to type this as a `Redux.Reducer`:

https://github.com/reduxjs/redux/blob/344d0e2347b3fc2221e626d495f4a12ac95907f0/index.d.ts#L61-L64

Reply:
yep. we should rely on defined types as much as possible, as they are usually more complicated but make up for it in developer-experience, e.g. by removing props the way that `connect()` in `react-redux` does

all things considered, this is going to be one of our more complicated type signatures because of how it's put together and how generic it is. what can we preserve from the underlying code? if we had presented `key` as a single string we could actually track which keys are in state at any given point in time in the code. we might be able to keep this with our joined string array though the typing would awfully complicated as we'd have to perform computation with the types.

I'd like to play around with this and see what I can come up with. In the meantime, we can actually relax or simplify our constraints by using the existing `Reducer` type. notice that `( state = false, action ) => true` doesn't match the existing definition because `false` isn't `object`

```ts
export const addReducerToStore = <T extends Reducer>( store: object ) => (
	key: string[],
	reducer: T
): Promise<void> => {
	…
}
```

Reply:
Guess I stumbled onto a difficult first TypeScript experience 😅 Made the change you suggested.

Reply:
hope you have a quick ramp-up to getting comfortable with more complicated types. I'm sure you will do fine.